src/core/utils/uint256.c

```diff
@@ -134,20 +134,26 @@ uint256_t *uint256_from_hex_str(char const *str) {
   }
 
   // create temporary string with maximum 256 bit hex length (4 * uint64_t)
-  char str_max_temp[str_max_len];
```

Comment:
MSVC doesn't support the variable-length array(VLA), this feature is important for embedded projects to avoid problems of dynamic memory allocation. 
we can use the compiler's predefined macro to solve this issue

```c
#if defined(_MSC_VER)
// non-VAL implementation for MS Visual-Studio
#else 
// the original VLA implementation
#endif 
```

Reply:
I have changed the code: now using compiler's predefined macro for solving the issue of variable-length array